tests/test_mlctx.py

```diff
@@ -2,16 +2,16 @@
 from xdsl.dialects.builtin import UnregisteredAttr, UnregisteredOp
 
 from xdsl.ir import MLContext, TypeAttribute, ParametrizedAttribute
-from xdsl.irdl import irdl_op_definition, irdl_attr_definition, Operation
+from xdsl.irdl import irdl_op_definition, irdl_attr_definition, IRDLOperation
 
 
 @irdl_op_definition
-class DummyOp(Operation):
+class DummyOp(IRDLOperation):
     name = "dummy"
 
 
 @irdl_op_definition
-class DummyOp2(Operation):
+class DummyOp2(IRDLOperation):
     name = "dummy2"
```

Comment:
For me, that's one of the symptoms. This test works jut fine without `@irdl_op_definition`, because it works on a different level. (I'm not sure if I want to call higher or lower here... But this doesn't care about IRDL in itself).
Should we just make those `Operation`s?

Reply:
Yes, I think they should be.

In the future I would even argue that we should remove IRDL entirely from the tests of ir/mlctx/printer/... to split the core data structures and IRDL as much as possible.

Reply:
Done for test_mlctx, and in parts for printer/ir
In parts, because removing the rest unveil new refactor opportunities... Follow-up PR material :slightly_smiling_face: